nilearn/glm/tests/test_thresholding.py

```diff
@@ -620,46 +620,25 @@ def test_threshold_stats_img_surface_output_threshold_0(surf_img_1d):
     )
 
 
-@pytest.mark.parametrize("threshold", [3.0, 2.9, DEFAULT_Z_THRESHOLD])
-@pytest.mark.parametrize("height_control", [None, "bonferroni", "fdr", "fpr"])
-def test_deprecation_threshold(surf_img_1d, height_control, threshold):
-    """Check warning thrown when threshold==old threshold.
+def test_deprecation_threshold(surf_img_1d):
+    """Check deprecation warning for default threshold.
 
     # TODO (nilearn >= 0.15.0)
     # remove
     """
-    with warnings.catch_warnings(record=True) as warning_list:
-        threshold_stats_img(
-            surf_img_1d, height_control=height_control, threshold=threshold
-        )
-
-    n_warnings = len(
-        [x for x in warning_list if issubclass(x.category, FutureWarning)]
-    )
-    if height_control is None and threshold == 3.0:
-        assert n_warnings == 1
-    else:
-        assert n_warnings == 0
+    with pytest.warns(FutureWarning, match="From nilearn version>=0.15"):
+        threshold_stats_img(surf_img_1d, height_control=None, threshold=3.09)
 
 
 @pytest.mark.slow
-@pytest.mark.parametrize("threshold", [3, 3.0, 2.9, DEFAULT_Z_THRESHOLD])
 def test_deprecation_threshold_cluster_level_inference(
-    threshold, img_3d_rand_eye, surf_img_1d
+    img_3d_rand_eye, surf_img_1d
 ):
     """Check cluster_level_inference warns when threshold==old threshold .
 
     # TODO (nilearn >= 0.15.0)
     # remove
     """
     for stat_img in [img_3d_rand_eye, surf_img_1d]:
-        with warnings.catch_warnings(record=True) as warning_list:
-            cluster_level_inference(stat_img, threshold=threshold)
-
-        n_warnings = len(
-            [x for x in warning_list if issubclass(x.category, FutureWarning)]
-        )
-        if threshold == 3.0:
-            assert n_warnings == 1
-        else:
-            assert n_warnings == 0
```

Comment:
same here: the warning that the default will be changed should only be thrown when the default threshold value is used